scripts/prepare_network.py

```diff
@@ -59,18 +59,72 @@
 import logging
 import os
 import re
+from zipfile import ZipFile
 
+import country_converter as coco
 import numpy as np
 import pandas as pd
 import pypsa
-from _helpers import configure_logging
+import requests
+from _helpers import configure_logging, create_country_list
 from add_electricity import load_costs, update_transmission_costs
+from config_osm_data import continent_regions, world_iso
 
 idx = pd.IndexSlice
 
 logger = logging.getLogger(__name__)
 
 
+def download_emission_data():
+    try:
+        url = "https://jeodpp.jrc.ec.europa.eu/ftp/jrc-opendata/EDGAR/datasets/v60_GHG/CO2_excl_short-cycle_org_C/v60_GHG_CO2_excl_short-cycle_org_C_1970_2018.zip"
+        with requests.get(url) as rq:
+            with open("data/co2.zip", "wb") as file:
+                file.write(rq.content)
+        rootpath = os.getcwd()
+        file_path = os.path.join(rootpath, "data/co2.zip")
+        with ZipFile(file_path, "r") as zipObj:
+            zipObj.extract(
+                "v60_CO2_excl_short-cycle_org_C_1970_2018.xls", rootpath + "/data"
+            )
+        os.remove(file_path)
+        return "v60_CO2_excl_short-cycle_org_C_1970_2018.xls"
+    except:
+        logger.warning(f"Failed download resource from '{url}'.")
+        return False
+
+
+def emission_extractor(filename):
+    country_names = create_country_list(snakemake.config["countries"])
+    # data reading process
+    datapath = os.path.join(os.getcwd(), "data", filename)
+    emission_of_countries = []
+    df = pd.ExcelFile(datapath)
+    df = pd.read_excel(df, sheet_name="v6.0_EM_CO2_fossil_IPCC1996", skiprows=8)
+    df.columns = df.iloc[0]
+    # NEPAL
+    df.loc[3285, "Y_1990"] = df.loc[3285, "Y_1988"]
+    # GREENLAND
+    df.loc[2037, "Y_1990"] = df.loc[2037, "Y_2004"]
+    # GHANA
+    df.loc[1099, "Y_1990"] = df.loc[1099, "Y_1986"]
+    # TURKS AND CAICOS ISLANDS
+    df.loc[491, "Y_1990"] = df.loc[491, "Y_1995"]
```

Comment:
Could you explain why we need this? Is the data complete?

Reply:
1990 emission data is missing in those countries. Therefore I have filled by the values before 1990 (if any) otherwise used most recent year.

Reply:
Are there other columns that are fully filled?

Reply:
Yes, depending on the country, emission values in other years available and I have used them to fill missing value at 1990 (preferable before 1990).